### PR TITLE
feat(APIApplicationCommand): add `channel_types` field to channel options

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds the `channel_types` option to command options that use channel selection.

**Reference Discord API Docs PRs or commits:**
- https://github.com/discord/discord-api-docs/pull/3825
- https://github.com/discordjs/discord.js/pull/6640

### CODE REVIEW

File: payloads/v9/_interactions/_applicationCommands/chatInput.ts [hunk 1]
```diff
@@ -54,6 +55,17 @@ export interface APIApplicationCommandArgumentOptions extends Omit<APIApplicatio
...
 }
 
+/**
+ * This type is exported as a way to make it stricter for you when you're writing your commands
+ *
+ * In contrast to `APIApplicationCommandSubCommandOptions` and `APIApplicationCommandArgumentOptions`,
+ *  these types cannot have an `options` array, or a `choices` array, but they can have a `channel_types` one.
+ */
+export interface APIApplicationCommandChannelOptions extends Omit<APIApplicationCommandOptionBase, 'type'> {
+	type: ApplicationCommandOptionType.Channel;
```
Comment (vladfrangu) [inline payloads/v9/_interactions/_applicationCommands/chatInput.ts:65]: Not related to this line but please remove line 16 in this file! :D

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Small change needed, as well as running the deno hooks, otherwise LGTM

File: payloads/v9/_interactions/_applicationCommands/chatInput.ts (hunk 1 above)
Comment (suneettipirneni) [reply payloads/v9/_interactions/_applicationCommands/chatInput.ts:64]: Gotcha, removed it from the deno version as well.

**Comment by vladfrangu** [comment]
Please let the `build:deno` script/hook do the deno changes 🙏

**Comment by suneettipirneni** [comment]
Ohhh that's automated... should I revert my last commit?

**Comment by vladfrangu** [comment]
No need to revert commits, just run the script and commit it's results!

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by suneettipirneni** [comment]
PR has been merged upstream

**Review by vladfrangu — CHANGES_REQUESTED** [review]
v8 pls

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — CHANGES_REQUESTED** [review]
In payloads/v8/_interactions/_applicationCommands/chatInput.ts, you forgot to remove line 16

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**